source/components/TextArea.js

```diff
@@ -95,66 +107,74 @@ class TextArea extends Component {
     }
   }
 
-  focus = () => this.textareaElement.focus();
+  focus = () => {
+    const { textareaElement } = this;
+    if (textareaElement && textareaElement.current) {
+      textareaElement.current.focus();
+    }
+  }
 
-  onChange = event => {
+  onChange = (event: SyntheticInputEvent<>) => {
     const { onChange, disabled } = this.props;
     if (disabled) return;
 
     if (onChange) onChange(this._processValue(event.target.value), event);
   };
 
-  _setError = error => this.setState({ error });
+  _setError = (error: string) => this.setState({ error });
 
-  _processValue(value) {
+  _processValue(value: string) {
     return flow([
       this._enforceStringValue,
       this._enforceMaxLength,
       this._enforceMinLength
     ]).call(this, value);
   }
 
-  _enforceStringValue(value) {
-    if (!isString(value)) throw 'Values passed to Input::onChange must be strings';
+  _enforceStringValue(value: string) {
+    if (!isString(value)) {
+      throw new Error('Values passed to TextArea::onChange must be strings');
+    }
     return value;
   }
 
-  _enforceMaxLength(value) {
+  _enforceMaxLength(value: string) {
     const { maxLength } = this.props;
     const isTooLong = maxLength != null && value.length > maxLength;
     return isTooLong ? value.substring(0, maxLength) : value;
   }
 
-  _enforceMinLength = value => {
+  _enforceMinLength(value: string) {
     const { minLength } = this.props;
     const isTooShort = minLength != null && value.length < minLength;
 
     if (isTooShort) {
-      this._setError(`Please enter a valid input`);
+      this._setError('Please enter a valid input');
     } else if (this.state.error !== '') {
-      this._setError(null);
+      this._setError('');
     }
 
     return value;
-  };
+  }
 
-  _handleAutoresize = () => {
+  _handleAutoresize() {
     const { textareaElement } = this;
 
-    if (!textareaElement) return;
-
-    // compute the height difference between inner height and outer height
-    const style = getComputedStyle(textareaElement, null);
-    const heightOffset =
-      style.boxSizing === 'content-box'
-        ? -(parseFloat(style.paddingTop) + parseFloat(style.paddingBottom))
-        : parseFloat(style.borderTopWidth) + parseFloat(style.borderBottomWidth);
-
-    // resize the input to its content size
-    textareaElement.style.height = 'auto';
-    textareaElement.style.height = `${textareaElement.scrollHeight +
-      heightOffset}px`;
-  };
+    if (textareaElement && textareaElement.current) {
```

Comment:
Same here … instead of conditionals please use guard clauses (e.g: `if (!textareaElement) return;` like before) which return early in the beginning of the function.

Reply:
Similarly, flow complains here because with React.createRef, you call an ref's methods off the "current" property, so the warning complains that I can't only check for textareaElement because current may be empty even though textareaElement exists.